### DISCUSSION TITLE
How Positron projects work; how they differ from RStudio projects; how they differ from Code OSS / VScode folders

### DISCUSSION THREAD

**Discussion by mark-andrews** [posit-dev/positron#5425] [Q&A] [answered]
In RStudio, a project is basically a folder with a `.Rproj` file in it, and when we open a project in RStudio, it sets the working directory to the folder, which is the primary reason for using RStudio projects.
Positron projects do not, at least in my experience thus far, create files analogous to RStudio's `.Rproj` files. They do set the working directory appropriately, however, when we open or switch to the project.
But it seems like a "folder" in the Code OSS / VScode sense, which exist in Positron in the exact same way as other Code OSS IDEs, does the exact same thing as Positron projects. When I open a folder in Positron, it switches my working directory. Also, that folder is listed, along with projects, in the Recently Opened list (upper right).
So is there a difference between a folder and a project in Positron?
A related issue is that if I do File / New Project in Positron, I can only create a new directory for that project. I can not use an existing directory. In RStudio by contrast, when I do File / New Project, I get a dialog box allowing me to create a new directory or use an existing directory. Is there a reason why we can't or shouldn't or need not do the same thing in Positron?

**Comment by juliasilge** [comment, marked as answer]
Thanks for the question @mark-andrews! Writing some better documentation about workspaces/folders/projects is one of the items we have outlined in https://github.com/posit-dev/positron/issues/4157 and it's helpful to know the relative importance of this problem.

In some ways, the situation in Positron is simpler than RStudio; no special files! Positron workspaces are most similar to VS Code/Code OSS workspaces, and are approximately the same conceptually as an RStudio project except with no `.Rproj` file, and _all_ of these align in most cases with a literal folder.

The "Workspaces: New Project" command creates a new workspace; we've used the word "project" here to try to indicate that this isn't quite the same as "Workspaces: New Folder" (this New Project Wizard will set up certain files and, for example, a Python venv for you) but we understand it is confusing. We have https://github.com/posit-dev/positron/issues/4787 where we are thinking about how to be more clear in our use of these terms.

You are correct that the New Project Wizard in Positron will only create an entirely new directory and will not make changes to an existing directory. If you have an existing directory you want to work in, just go ahead and open it up; it is then a Positron workspace (no need to go through the New Project Wizard or explicitly make it a workspace/project).

**Comment by mark-andrews** [comment]
Hi @juliasilge Thanks a lot for the explanation.
